### PR TITLE
Breadcrumbs and changed add data icons

### PR DESCRIPTION
**Pull request by ClockworkOnion**
Changes from https://github.com/geo-engine/geoengine-ui/pull/423 reapplied into the new file/folder structure

### CODE REVIEW

**Comment by ClockworkOnion** [comment]
[Breadcrumbs fürs Layerlisting](https://trello.com/c/ePrStiCR/894-breadcrumbs-f%C3%BCrs-layerlisting)

File: projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts [hunk 1]
```diff
@@ -10,47 +10,96 @@ import {CONTEXT_TOKEN, LayerCollectionListComponent} from '../layer-collection-l
...
+    @ViewChild('scrollElement', {read: ElementRef}) public scrollElement!: ElementRef<any>;
+
     constructor() {
         this.setPortal(undefined);
     }
 
+    scrollToRight(): void {
+        setTimeout(() => {
+            // wait until breadcrumbs are re-rendered before scrolling
+            this.scrollElement.nativeElement.scrollLeft += Number.MAX_SAFE_INTEGER;
```
Comment (michaelmattig) [inline projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts:30]: this doesn't work, at least in firefox it doesn't scroll.

File: projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts [hunk 2]
```diff
@@ -10,47 +10,96 @@ import {CONTEXT_TOKEN, LayerCollectionListComponent} from '../layer-collection-l
...
+    updateLayerView(): void {
+        const currentTrail = this.allTrails[this.selectedCollection];
+        this.displayedTrail = currentTrail;
+        const lastId = currentTrail[currentTrail.length - 1];
+        this.setPortal(lastId);
+    }
+
+    onBreadCrumbClick(index: number): void {
+        // Creates and appends a new crumbtrail, then moves forward to it
+        if (index === this.displayedTrail.length - 1) return;
```
Comment (michaelmattig) [inline projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts:78]: please add curly braces around the return

File: projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts [hunk 3]
```diff
@@ -10,47 +10,96 @@ import {CONTEXT_TOKEN, LayerCollectionListComponent} from '../layer-collection-l
...
+        const newTrail = this.displayedTrail.map((x) => Object.assign({}, x)).slice(0, index + 1);
+        this.allTrails.push(newTrail);
+        this.selectedCollection = this.allTrails.length - 2;
+        this.forward();
+    }
+
+    navigateToRoot(): void {
+        if (this.selectedCollection === -1) return;
```
Comment (michaelmattig) [inline projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts:86]: please add curly braces around the return

File: projects/core/src/lib/datasets/add-data/add-data.component.ts [hunk 4]
```diff
@@ -56,7 +55,7 @@ export class AddDataComponent implements OnInit {
         return {
             name: 'Layers',
             description: 'Collection of existing layers',
-            iconSrc: createIconDataUrl('Layer Collection'),
+            icon: 'sort',
```
Comment (michaelmattig) [inline projects/core/src/lib/datasets/add-data/add-data.component.ts:58]: wasn't this supposed to be `layers`?

File: projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts (hunk 2 above)
Comment (ClockworkOnion) [reply projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts]: Added them in 0a65ceefb42d3a9b9bc603e0ea676a8a9eb358fb

File: projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts (hunk 3 above)
Comment (ClockworkOnion) [reply projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts]: Added them in 0a65ceefb42d3a9b9bc603e0ea676a8a9eb358fb

File: projects/core/src/lib/datasets/add-data/add-data.component.ts (hunk 4 above)
Comment (ClockworkOnion) [reply projects/core/src/lib/datasets/add-data/add-data.component.ts]: Oops, fixed that in 0a65ceefb42d3a9b9bc603e0ea676a8a9eb358fb

File: projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts (hunk 1 above)
Comment (ClockworkOnion) [reply projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts]: Ideally, we could use the full width of the breadcrumb area here, as if it hadn't a scrollbar. But I can only find the width of the element after the scrollbar is applied, which is of course a fixed value and not large enough.

Although it's a magic number, I used 900000 now, which should be high enough in any case and tested it in Firefox and Safari as well.  0a65ceefb42d3a9b9bc603e0ea676a8a9eb358fb

Comment (ClockworkOnion) [reply projects/core/src/lib/layer-collections/layer-collection-navigation/layer-collection-navigation.component.ts]: It works on Chrome, Firefox, Safari and Edge

**Review by michaelmattig — APPROVED** [review]

**Pull request merged**